storage/src/test/java/io/atomix/storage/buffer/BufferTest.java

```diff
@@ -671,4 +672,121 @@ public void testCapacity0Write() {
     assertEquals(10, buffer.readLong(0));
   }
 
+  @Test
+  public void shouldWriteByteBuffer() {
+    // given
+    final Buffer dest = createBuffer(0, 1024);
+    final ByteBuffer src = ByteBuffer.allocateDirect(2 * Long.BYTES);
+    src.putLong(1L).putLong(Long.BYTES, 2L);
+
+    // when
+    dest.write(src);
+
+    // then
+    // check that we wrote that we expect
+    assertEquals(2L, dest.readLong(0));
```

Comment:
shouldn't this be 1?

Reply:
The position of `src` is at `Long.BYTES` when we write, and writing will start at the current position, so it writes the second long.

Reply:
this seems weird :sweat_smile: